Shop-Homework/src/homework/shop/PurchaseManager.java

```diff
@@ -0,0 +1,43 @@
+package homework.shop;
+
+import static homework.shop.Product.AGE_RESTRICTION_ADULT;
+import static homework.shop.Product.AGE_RESTRICTION_TEENAGER;
+
+/**
+ * this class contains the processPurchase method that handles purchases
+ */
+public class PurchaseManager {
+
+    public static void processPurchase(Product product, Customer customer) {
```

Comment:
in metoda trebuie sa scazi din bugetul constumer-ului pretul produsului si din quantity-ul produsul sa se scada 1.